### DISCUSSION TITLE
RAG only working with Claude?

### DISCUSSION THREAD

**Discussion by tobwen** [open-webui/open-webui#5202] [Q&A]
Before posting a bugreport: Is it a known problem that documents can only be read from Claude, but not OpenAI? It neither works for single, nor for multi-models:

![image](https://github.com/user-attachments/assets/37532d82-9ebf-4aff-ade0-bab18c7a875b)

**Comment by thiswillbeyourgithub** [comment]
I never got RAG to work. When using langfuse I see that the context parsed is apparenty empty text, even though tika succesfully parsed the content according to the log.

I asked over [there in an issue](https://github.com/open-webui/open-webui/issues/5034) to error out when a rag template sees an empty string but got closed without explanation.

Sorry for the ping @tjbck but can you reevaluate my issue? Here's a particularly striking but common situation:
- user sends a pdf about a company's history, say samsung
- user asks "what's the  ontext of creation of Samsung?"
- user expects to be using RAG, but actually it has the same bug as me and tobwe so user will only get info from the LLM's knowledge. But user wimm not know that, because open-webui does not check that we're not doing RAG on an empty string.

I've seen this many times : users using rag but asking question where you can't know if the LLM actually read the document.

My proposed solution is simple: raise an exception if the rag template does not contain "[context]", or if the context.strip() is empty.


Related :
- #5002 where @qiulang seems to have the same issue
- #5050 
- #5291 


Fwiw I tried every combination of RAG settings and stilm have no idea what causes it. But the lack of error when something clearly went wrong (RAG on an empty text) is certainly not helping. I tried with and without tika, many different pdfs, txt files, etc. In any case the parsing worked according the the logs.
